### ISSUE TITLE
QGIS r.sun.insoltime is CPU locked? 

### ISSUE THREAD

**Issue by Case2903** [OSGeo/grass#3487]
Hello,

I am trying to run r.sun for a large project, the process is very slow and it runs on CPU only. Is there a way to speed up the process? maybe enabling GPU for processing. is this possible or is this kind of algorithm CPU locked?

**Comment by neteler** [comment]
Yes, CPU parallelization based on openMP has been available for some time in `r.sun`.

However, in the QGIS wrapper script `r.sun.insoltime.txt`  the parameter "nproc" is yet missing. It would be sufficient to (locally) add this line, so in your local installation you could simply add it:

```
*QgsProcessingParameterNumber|nprocs|Number of threads which will be used for parallel computing|QgsProcessingParameterNumber.Integer|1|True|1|None
```

(I have tested this change successfully on my local Fedora box).

The sustainable solution would be to add this parameter in QGIS via a pull request, editing the file:

https://github.com/qgis/QGIS/blob/master/python/plugins/grassprovider/description/r.sun.insoltime.txt
